Enable epoch boundary hook in conformance

carlostome opened the pull request:
# Description

This PR enables `_epochBoundaryConformanceHook` in conformance testing. 

Enabling the hook causes 7 test failures (against the updated fls), and thus this PR also disables those test selectively. 

To that end it adds a new combinator `disableInConformanceIt` to replace `hspec`'s `it` in those test for which we want to disable conformance testing. For instance:

```haskell
      -- https://github.com/IntersectMBO/formal-ledger-specifications/issues/916
      -- TODO: Re-enable after issue is resolved, by removing this override
      disableInConformanceIt "Delegate vote and undelegate after delegating to some stake pools" $ do
```

In addition, this PR:

1. fixes the implementation of `disableImpInitPostEpochBoundaryHook`
2. updates fls
3. restructures a couple of (conformance failing) test following the aforementioned structure

# Checklist

- [x] Commits in meaningful sequence and with useful messages.
- [ ] Tests added or updated when needed.
- [ ] `CHANGELOG.md` files updated for packages with externally visible changes.  
      **NOTE: _New section is never added with the code changes._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd)).
- [ ] Versions updated in `.cabal` and `CHANGELOG.md` files when necessary, according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] Version bounds in `.cabal` files updated when necessary.  
      **NOTE: _If bounds change in a cabal file, that package itself must have a version increase._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process)).
- [x] Code formatted (use `scripts/fourmolize.sh`).
- [x] Cabal files formatted (use `scripts/cabal-format.sh`).
- [x] CDDL files are up to date (use `scripts/gen-cddl.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) updated (use `scripts/gen-hie.sh`).
- [x] Self-reviewed the diff.

carlostome committed:
Fix implementation of disableImInitPostEpochBoundaryHook

carlostome committed:
Make mismatches in conformance tests explicit

carlostome committed:
Update fls

carlostome committed:
Enable epoch boundary hook

lehins reviewed (approved):
Very nice! Thank you!

lehins commented:
Looks like SRP update is not pointing to a commit on master which needs to be fixed before we can merge this PR

carlostome committed:
Add disableInConformanceIt

carlostome commented:
I incorporated the comments from yesterday's discussion:
- Fixed the SRP
- Added a combinator `disableInConformanceIt` to replace `it`, which disables the conformance hooks and appends the text "[disabled in conformance]"

Regarding the long diffs. 

The way conformance was disabled was by using the hooks in the line before the test, and hence the test body was indented:
```haskell
disableImpInitPostSubmitTxHook $ do
  it "test" $ do
    test body
```
Now `disableInConformanceIt` replaces both the hook and the description and thus the test body has to be disindented one level. On the upside, when we fix the tests in conformance and reenable them, the diffs will be very small.

carlostome commented on eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/EnactSpec.hs:
```diff
@@ -151,8 +153,12 @@ treasuryWithdrawalsSpec =
-      whenPostBootstrap $ do
+    -- https://github.com/IntersectMBO/formal-ledger-specifications/issues/912
+    -- TODO: Re-enable after issues are resolved, by removing this override
+    disableInConformanceIt
+      "Withdrawals exceeding treasury submitted in several proposals within the same epoch"
```
it would be nicer if this was one line, but formolu complains

lehins reviewed (approved):
:+1:
